geps/gep-1897.md

```diff
@@ -125,7 +126,235 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+// Note: there is no Override or Default policy configuration.
+type TLSConnectionPolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // Services are the only valid API target references.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+
+    // TLS contains TLS connection policy configuration.
+    TLS *TLSConnectionPolicyConfig `json:”tls”`
+}
+
+// TLSConnectionPolicyConfig contains TLS connection policy configuration.
+type TLSConnectionPolicyConfig struct {
+    // TrustedCACertRefs contains one or more references to
+    // Kubernetes objects that contain TLS certificates, which are
+    // used to establish a TLS handshake between the gateway and
+    // backend pod.
+    //
+    // A single TrustedCACertRef to a Kubernetes Secret has "Core"
+    // support.  Implementations MAY choose to support attaching
+    // multiple certificates to a backend, but this behavior is 
+    // implementation-specific.
+    //
+    // References to a resource in a different namespace are 
+    // invalid.
+    // 
+    // This field is required for any TLSConnectionPolicyConfig.
+    // 
+    // Support: Core - A single reference to a Kubernetes Secret of type kubernetes.io/tls
+    //
+    // Support: Implementation-specific (More than one reference or other resource types)
+    //
+    // +kubebuilder:validation:MaxItems=64
+    // +kubebuilder:validation:MinItems=1
+    TrustedCACertRefs []SecretObjectReference `json:”trustedCACertRefs”`
+
+    // Port is the network port that the implementation watches to
+    // know if the connection should be TLS and the targetRef’s
+    // certificate should be validated by the certs in TrustedCACertRefs
+    // If empty, then all ports for the targetRef are watched to know
+    // if the connection should be TLS, the targetRef’s certificate
+    // should be validated by the certs in TrustedCACertRefs, and a
+    // status delivered in the response for validation failures.
+    Port PortNumber `json:port,omitempty`
+}
+
+// TLSConnectionPolicyStatus defines the observed state of TLSConnectionPolicy.
+type TLSConnectionPolicyStatus struct {
+    // Conditions describe the current conditions of the TLSConnectionPolicy.
+    //
+    // Implementations should prefer to express TLSConnectionPolicy
+    // conditions using the `TLSConnectionPolicyConditionType` and
+    // `TLSConnectionPolicyConditionReason` constants so that 
+    // operators and tools can converge on a common vocabulary to 
+    // describe TLSConnectionPolicy state.
+    // Known condition types are:
+    // 
+    // * “Accepted”
+    // 
+    // +optional
+    // +listType=map
+    // +listMapKey=type
+    // +kubebuilder:validation:MaxItems=8
+    // +kubebuilder:default={type: "Accepted", status: "Unknown", reason:"Pending", message:"Waiting for validation", lastTransitionTime: "1970-01-01T00:00:00Z"}
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
Assuming multiple implementations running in the cluster,  will they start conflicting with each other when updating the status of the resource?

Reply:
It's an interesting question.  I haven't seen the same question applied to other proposals, is it pertinent here because TLSConnectionPolicy isn't namespaced?  I wouldn't expect multiple implementations to interfere with each other, @robscott @shaneutt @youngnick are you aware if there's a GEP addressing multiple-implementations co-existing on a cluster?  I'd expect that each implementation knows how to access only those objects that it created, but I don't know for sure.

Reply:
Yes, that's a very good point. We probably need to introduce some form of nesting here, similar to [Route status](https://github.com/kubernetes-sigs/gateway-api/blob/a33a934af9ec6997b34fd9b00d2ecd13d143e48b/apis/v1beta1/shared_types.go#L361) where each set of conditions is per-Parent. I'm not sure what we'd call that, maybe `TargetParents` instead of just `Parents`? 

Open to ideas, but agree that it's a gap in our policy attachment model. Right now it doesn't really define how status should be handled for policies that are implemented by multiple controllers.

Reply:
In general, assuming multiple controllers feels like a best practice for Gateway API resources _unless_ the resource in question has some field that effectively limits it to a single controller (e.g. `gatewayClassName` on Gateway). I'm in favor of making the change in this GEP specifically as well as a general change to the policy attachment model

Reply:
Added to Open Questions section.